Dockerfile

```
@@ -0,0 +1,8 @@
+FROM node:12-alpine
+WORKDIR /dist
+COPY package.json ./
+COPY package-lock.json ./
+RUN npm install
+COPY . .
+EXPOSE 3000
+CMD [ "node", "server.js" ]
```

Так ты создаешь папку dist в корне и ставишь в нее модули, а затем копируешь все содержимое dist из проекта туда же, и потом это сервится из контейнера (npm run build не происходит, установленные модули просто лежат). Чтобы потом сделать актуальный (новый) образ, после того, как будут внесены правки в проект, нужно будет каждый раз локально делать новый билд, а уже потом новый образ и контейнер (нужно следить чтобы "локальный" dist был всегда в актуальном состоянии)
Но dist должен быть в dockerignore по-хорошему, его не нужно копировать, нужно скопировать src проекта, и делать билд "в образе", предварительно закешировав слой модулей (это у тебя сделано)
```suggestion
FROM node:12-alpine
WORKDIR /app // это не связано с локальной папкой в проекте, это директория которая будет в контейнере
COPY package.json ./
COPY package-lock.json ./
RUN npm install
COPY . .
RUN npm run build
EXPOSE 3000
CMD [ "node", "server.js" ]
```